Dockerfile

```diff
@@ -0,0 +1,37 @@
+# use CUDA_TAG to select the image version to use
+# see https://hub.docker.com/r/nvidia/cuda/
+#
+# For example to create a ubuntu 16.04 with cuda 8.0 for development, use
+# docker build --build-arg CUDA_TAG=8.0-devel --tag popsift .
```

Comment:
```
# For example to create a ubuntu 16.04 with cuda 8.0 for development, use:
# docker build --build-arg CUDA_TAG=8.0-devel-ubuntu16.04 --tag popsift .
# Or to create a centos7 with cuda 9.2, use:
# docker build --build-arg CUDA_TAG=9.2-devel-centos7` --tag popsift .
```

Reply:
In principle you are right, but I don't think it works for CentOS as we use `apt-get` to get some of the dependencies and CentOS uses `yum`
I don't know, if we want to make it platform independent, we have either to install each dependency by hand or use CUDA_TAG to decide to use either `apt` or `yum`

Reply:
@fabiencastan Also now I realize that this could only work on a ubuntu > 14.4 because, for example, the default version of cmake of 14.04 does not meet the minimum requirements.

That been said, I think that in principle you can run an ubuntu image on a centos host or on a lower version of ubuntu, the only thing that matters is that the host's nvidia runtime/driver matches (>=) the cuda version of the image. 
So I don't know if it is really interesting to have a script that can be build on any machine: ideally a good idea but to make it you need a lot more scripting than this or a cmake script like alicevision.